src/project-css/app/Helpers/CollectionHelper.php

```diff
@@ -0,0 +1,193 @@
+        if (array_key_exists('cmsId', $data)) {
+          $thisClctn->cmsId = $data['cmsId'];
+        }
+
+        $thisClctn->save();
+
+        // create folder in storage that will contain any additional files associated to the collection
+        if (Storage::exists($thisClctn->clctnName) == FALSE) {
+          Storage::makeDirectory($thisClctn->clctnName, 0775);
+        }
+
+        return $thisClctn;
+     }
+
+     public function update($data) {
+      // find the collection
+      $thisClctn = Collection::findorFail($data['id']);
+
+      if ($thisClctn->clctnName != $data['name']) {
+        // Rename Storage Folder
+        if ((Storage::exists($data['name']) == FALSE) && (Storage::exists($thisClctn->clctnName))) {
+          Storage::move($thisClctn->clctnName, $data['name']);
+        }  
+
+        // Set new Collection Name
+        $thisClctn->clctnName = $data['name'];
+      }
+
+      if ($thisClctn->isCms != $data['isCms']) {
+        // Set CMS State
+        $thisClctn->isCms = $data['isCms'];
+      }
+
+      // Save Updated items
+      $thisClctn->save();            
+     }
+
+     public function disable($name) {
+        // verify collection exists
+        if ($this->isCollection($name)) {
+          // find the collection
+          $thisClctn = Collection::where('clctnName', $name)->first();
+
+          // reset collection to disabled
+          $this->updateCollectionFlag($thisClctn->id, false);
+
+          return true;
+        }
+
+        return false;
+    }
+
+    public function enable($name) {
+      // verify collection exists
+      if ($this->isCollection($name)) {
+        // find the collection
+        $thisClctn = Collection::where('clctnName', $name)->first();
+
+        // reset collection to enabled
+        $this->updateCollectionFlag($thisClctn->id, true);
+
+        return true;
+      }
+
+      return false;
+     }
+
+     public function isCollection($name) {
+        if (Collection::where('clctnName', '=', $name)->count() == 1) {
+          return true;
+        }
+        return false;
+     }
+
+     public function setCMS($name, $option = false) {
+        // find the collection
+        $thisClctn = Collection::where('clctnName', $name)->first();
+
+        // Set isCms
+        $thisClctn->isCms = $option;
+
+        // Save Updated items
+        $thisClctn->save();  
+     }
+
+
+     // check and see if collection has Tables
+     // associated to it.
+     public function hasTables($name) {
+      if ($this->isCollection($name) == false) {   
+        return false;
+      }
+        
+      // find the collection
+      $thisClctn = Collection::where('clctnName', $name)->first();
+
+      return $thisClctn->hasTables();
+     }
+
+    /**
+    * Returns true if files exist in collection. False otherwise.
+    *
+    * @param  string  $name    
+    * @return boolean
+    */         
+     public function hasFiles($name) {
+       if ($this->isCollection($name)) {
```

Comment:
You are compulsively checking this, maybe this class should have a constructor function that sets these so there isn't the need to make all of these checks.